markdown/epilogue.markdown

```diff
@@ -1,39 +1,151 @@
+you've tried a bit of problem solving yourself, and to provide others with a
+minimal script to demonstrate your problem and how you've tried to fix it.
+
+- **No:** "I need to generate a big array of random Gaussians. Can someone
+  help?"
+- **No:** "I have this huge library at https://github.com/ron_obvious, if you
+  look in the statistics library, there's a part that really needs random
+  Gaussians, can someone take a look???"
+- **Yes:** "I've been trying to generate a big list of random Gaussians like
+  so: `gauss = [np.random.randn()] * 10**5`. But when I compute `np.mean(gauss)`,
+  it's hardly ever as close to 0 as I expect. What am I doing wrong? The full
+  script ready for copy-paste is below.
+
+### GitHub
+
+We also talked in the preface about GitHub. All of the code that we discussed lives
+on GitHub:
+
+- https://github.com/numpy/numpy
+- https://github.com/scipy/scipy
+
+and others. When something isn't working as you expect, it could be a bug. If,
+after some investigation, you are convinced that you have indeed uncovered a
+bug, you should go to the "issues" tab of the relevant GitHub repository and
+create a new issue. This will ensure that the bug is on the radar of the
+developers of that library, and that it will (hopefully) be fixed in the next
+version. By the way, this advice also applies to "bugs" in the documentation:
+if something in a library's documentation isn't clear to you, file an issue!
+
+Even better than filing an issue is *submitting a pull request*. We can't cover
+the process here, but there are many books and resources out there to help you:
+- Anthony Scopatz and Katy Huff's [Effective Computation in
+  Physics](http://shop.oreilly.com/product/0636920033424.do) covers git and
+  GitHub, among many other topics in scientific computation.
+- [Introducing GitHub](http://shop.oreilly.com/product/0636920033059.do) by
+  Peter Bell and Brent Beer, covers GitHub in more detail.
+- [Software Carpentry](https://software-carpentry.org/) has git lessons, and
+  offers free workshops around the world throughout the year.
+- Based partly on those lessons, one of your authors has created a complete
+  tutorial on git and GitHub pull requests, [Open Source Science with Git and
+  GitHub](http://jni.github.io/git-tutorial/).
+- Finally, many open source projects on GitHub have a
+  ["CONTRIBUTING"](https://github.com/scikit-image/scikit-image/blob/master/.github/CONTRIBUTING.txt)
+  file, which contains a set of guidelines for contributing code to the project.
+
+So, you are not starved for help on this topic!
+
+What we would like to you is to encourage you to do this as often as you can,
+not only because you will help make these libraries better for all, but also
+because it is one of the best ways to develop your coding abilities. With every
+pull request you submit, you will get feedback about your code, helping you to
+improve. You'll also become more familiar with the GitHub contributing process
```

Comment:
There's some very significant common ground among projects... I prefer to keep it like this, for simplicity's sake.